__tests__/middleware.spec.ts

```diff
@@ -5,101 +5,94 @@
 import { verifyJwtToken } from '../lib/server/auth';
+import { getAuthTokenServer } from '@/utils/getAuthTokenServer';
+import { deleteCookie } from '@/utils/cookiesUtil';
 
 
 jest.mock('../lib/server/auth', () => ({
     verifyJwtToken: jest.fn(),
 }));
 
-const redirectSpy = jest.spyOn(NextResponse, 'redirect');
+jest.mock('@/utils/getAuthTokenServer', () => ({
+    getAuthTokenServer: jest.fn(),
+}));
 
-describe('middleware.ts', () => {
+jest.mock('@/utils/cookiesUtil', () => ({
+    deleteCookie: jest.fn()
+}))
 
-    const mockNextRequest = (url: string, cookies: Record<string, string> = {}) => {
-        return {
-            nextUrl: new URL(url, 'http://localhost:3000'),
-            cookies: {
-                get: (name: string) => (cookies[name] ? { value: cookies[name] } : undefined),
-                delete: jest.fn(),
-            },
-            headers: new Headers(),
-        } as unknown as NextRequest;
-    };
+describe('middleware.ts', () => {
+    let redirectSpy: jest.SpyInstance;
 
     beforeEach(() => {
+        redirectSpy = jest.spyOn(NextResponse, 'redirect');
         jest.clearAllMocks();
     });
 
-    it('should skip authentication for excluded paths', () => {
-        // @ts-ignore
-        const request = mockNextRequest('/api/setCookie');
-        const response = middleware(request);
-
-        expect(response).toEqual(NextResponse.next());
-    });
+    afterEach(() => {
+        redirectSpy.mockRestore();
+    })
 
-    it('should redirect to login if no token is found for protected routes', () => {
-        const request = mockNextRequest('/dmps/123');
-        middleware(request);
+    it('should redirect to login if no token is found for protected path', async () => {
+        (getAuthTokenServer as jest.Mock).mockResolvedValue(null);
+        const request = new NextRequest(new Request('http://localhost:3000/dmps/123'));
+        const response = await middleware(request);
 
-        expect(redirectSpy).toHaveBeenCalledWith(`${process.env.NEXT_PUBLIC_BASE_URL}/login`);
+        expect(redirectSpy).toHaveBeenCalledWith('http://localhost:3000/login');
     });
 
-    it('should delete token and redirect to login if token is invalid', () => {
-        const request = mockNextRequest('/dmps/123', { dmspt: 'invalid_token' });
+    it('should delete token and redirect to login if token is invalid', async () => {
+        (getAuthTokenServer as jest.Mock).mockResolvedValue('invalid_token');
+        const request = new NextRequest(new Request('http://localhost:3000/dmps/123'));
         (verifyJwtToken as jest.Mock).mockImplementation(() => false);
-        middleware(request);
+        await middleware(request);
 
-        expect(request.cookies.delete).toHaveBeenCalledWith('dmspt');
-        expect(redirectSpy).toHaveBeenCalledWith(`${process.env.NEXT_PUBLIC_BASE_URL}/login`);
+        expect(deleteCookie).toHaveBeenCalled();
+        expect(redirectSpy).toHaveBeenCalledWith('http://localhost:3000/login');
```

Comment:
Same as previous comment on hard-coded url. There are more in this document, but I'm not going to flag each individual one :)

Reply:
Yes, @andrewebdev as you mentioned above, I mocked the request and passed in the URL so that I could check the redirect url.